### PR TITLE
Improve fingerprint keyers - fixes #3282

### LINKED ISSUE #3282: Fingerprint functions should support Unicode whitespace and punctuation
Although the fingerprint keyers currently do diacritic folding for alphabetic characters, they don't correctly handle all Unicode whitespace and punctuation characters.

### Proposed solution

Both the FingeprintKeyer and NGramFingerprintKeyer should be extended to correctly handle all Unicode whitespace characters (e.g. em space, NBSP, ZWSP, etc) and punctuation.

Additionally, the (almost) duplicate code in the N-gram keyer should be removed and use the common methods from the fingerprint keyer to make maintenance easier and less bug prone.

### PR DESCRIPTION
**Pull request by tfmorris**
Fixes #3282 
- include all Unicode whitespace characters in splitting delimiter and don't strip controls which are whitespace (HT, LF, VT, FF, CR, NEL)
- add C1 controls to control characters that are stripped
- refactor string normalization to reduce redundancy between two keyers
- minor cleanups, simplifications, and performance optimizations

### CODE REVIEW

**Review by wetneb — APPROVED** [review]
Thanks a lot!

**Comment by tfmorris** [comment]
Sorry, I meant to clean up the LGTM warnings before this was merged.

Also, there was an open TODO that I should have highlighted in the pull request description:

https://github.com/tfmorris/OpenRefine/blob/45a3622dcf39c9d42470fb51fea23f4b9e682acb/main/src/com/google/refine/clustering/binning/FingerprintKeyer.java#L111-L113

asking whether we should be converting punctuation and controls to spaces instead of our current practice of removing them (effectively merging the strings on either side).

**Comment by tfmorris** [comment]
I pushed e809c707ffdffddc7ab87c3862e8a769200c7e78 to clean up the Javadoc and get rid of the LGTM errors.

**Pull request merged**
